pkg/odo/cli/url/create.go

```diff
@@ -204,52 +210,62 @@ func (o *URLCreateOptions) Complete(name string, cmd *cobra.Command, args []stri
 
 // Validate validates the URLCreateOptions based on completed values
 func (o *URLCreateOptions) Validate() (err error) {
-	// Check if exist
-	if experimental.IsExperimentalModeEnabled() && util.CheckPathExists(o.DevfilePath) {
+	if !util.CheckOutputFlag(o.OutputFlag) {
+		return fmt.Errorf("given output format %s is not supported", o.OutputFlag)
+	}
+
+	// if experimental mode is enabled, and devfile is provided.
+	errorList := make([]string, 0)
+	if o.isDevFile {
 		// check if a host is provided for route based URLs
-		if o.isRouteSupported && !o.wantIngress && len(o.host) > 0 {
-			return fmt.Errorf("host is not supported for URLs of Route Kind")
-		}
-		// if experimental mode is enabled, and devfile is provided.
-		// check if valid host is provided
-		if !pushtarget.IsPushTargetDocker() && len(o.host) <= 0 && (!o.isRouteSupported || o.wantIngress) {
-			return fmt.Errorf("host must be provided in order to create ingress")
+		if len(o.host) > 0 {
+			if o.urlType == envinfo.ROUTE {
+				errorList = append(errorList, "host is not supported for URLs of Route Kind")
+			}
+			if err := validation.ValidateHost(o.host); err != nil {
+				errorList = append(errorList, err.Error())
+			}
+		} else if o.urlType == envinfo.INGRESS {
+			errorList = append(errorList, "host must be provided in order to create URLS of Ingress Kind")
 		}
 		for _, localURL := range o.EnvSpecificInfo.GetURL() {
 			if o.urlName == localURL.Name {
-				return fmt.Errorf("the url %s already exists", o.urlName)
+				errorList = append(errorList, fmt.Sprintf("URL %s already exists", o.urlName))
 			}
 		}
 	} else {
 		for _, localURL := range o.LocalConfigInfo.GetURL() {
 			if o.urlName == localURL.Name {
-				return fmt.Errorf("the url %s already exists in the application: %s", o.urlName, o.Application)
+				errorList = append(errorList, fmt.Sprintf("URL %s already exists in application: %s", o.urlName, o.Application))
 			}
 		}
 	}
 	// Check if url name is more than 63 characters long
 	if len(o.urlName) > 63 {
-		return fmt.Errorf("url name must be shorter than 63 characters")
+		errorList = append(errorList, "URL name must be shorter than 63 characters")
 	}
 
-	if !util.CheckOutputFlag(o.OutputFlag) {
-		return fmt.Errorf("given output format %s is not supported", o.OutputFlag)
-	}
-	if !experimental.IsExperimentalModeEnabled() {
+	if !o.isExperimental {
 		if o.now {
-			err = o.ValidateComponentCreate()
-			if err != nil {
-				return err
+			if err = o.ValidateComponentCreate(); err != nil {
+				errorList = append(errorList, err.Error())
 			}
 		}
 	}
+
+	if len(errorList) > 0 {
+		for i := range errorList {
+			errorList[i] = fmt.Sprintf("\t- %s", errorList[i])
+		}
+		return fmt.Errorf("There are validation issues:\n%s", strings.Join(errorList, "\n"))
```

Comment:
How about something more simpler like `URL creation failed. Issues:` or `URL creation failed due to:` ?